src/osd/PeeringState.cc

```diff
@@ -403,17 +407,42 @@ void PeeringState::write_if_dirty(ObjectStore::Transaction& t)
   }
 }
 
+#ifndef WITH_SEASTAR
```

Comment:
why use negative condition? i propose
```suggestion
#ifdef WITH_SEASTAR
```